core/internal/flux_monitor_simulated_blockchain_test.go

```diff
@@ -0,0 +1,370 @@
+	}
+	mockServer := cltest.NewHTTPMockServerWithAlterableResponse(t, priceResponse)
+	defer mockServer.Close()
+
+	// When event appears on submissionReceived, flux monitor job run is complete
+	submissionReceived := make(chan *faw.FluxAggregatorSubmissionReceived)
+	subscription, err := fa.aggregatorContract.WatchSubmissionReceived(nil,
+		submissionReceived, []*big.Int{}, []uint32{},
+		[]common.Address{fa.nallory.From})
+	require.NoError(t, err, "failed to subscribe to SubmissionReceived events")
+	defer subscription.Unsubscribe()
+
+	// Create FM Job, and wait for job run to start (the above UpdateAnswer call
+	// to FluxAggregator contract initiates a run.)
+	buffer := cltest.MustReadFile(t, "testdata/flux_monitor_job.json")
+	var job models.JobSpec
+	require.NoError(t, json.Unmarshal(buffer, &job))
+	initr := &job.Initiators[0]
+	initr.InitiatorParams.Feeds = cltest.JSONFromString(t, fmt.Sprintf(`["%s"]`,
+		mockServer.URL))
+	initr.InitiatorParams.PollingInterval =
+		models.Duration(timeout)
+	initr.InitiatorParams.Address = fa.aggregatorContractAddress
+	j := cltest.CreateJobSpecViaWeb(t, app, job)
+	jrs := cltest.WaitForRuns(t, j, app.Store, 1) // Submit answer from
+	reportedPrice := jrs[0].RunRequest.RequestParams.Get("result").String()
+	assert.Equal(t, reportedPrice, fmt.Sprintf("%d", reportPrice),
+		"failed to report correct price to contract")
+	var receiptBlock uint64
+	select { // block until FluxAggregator contract acknowledges chainlink message
+	case log := <-submissionReceived:
+		receiptBlock = log.Raw.BlockNumber
+	case <-time.After(timeout):
+		t.Fatalf("chainlink failed to submit answer to FluxAggregator contract")
+	}
+	checkUpdateAnswer(t, answerParams{fa: &fa, roundId: roundId,
+		answer: processedAnswer, from: fa.nallory, isNewRound: false,
+		completesAnswer: true}, initialBalance, receiptBlock)
+
+	//- have the malicious node start the next round.
+	nextRoundBalance := initialBalance - fee
+	// Triggers a new round, since price deviation exceeds threshold
+	reportPrice = answer + 1
+	select {
+	case log := <-submissionReceived:
+		receiptBlock = log.Raw.BlockNumber
+	case <-time.After(2 * timeout):
+		t.Fatalf("chainlink failed to submit answer to FluxAggregator contract")
+	}
+	newRound := roundId + 1
+	processedAnswer = 100 * reportPrice
+	checkUpdateAnswer(t, answerParams{fa: &fa, roundId: newRound,
+		answer: processedAnswer, from: fa.nallory, isNewRound: true,
+		completesAnswer: false}, nextRoundBalance, receiptBlock)
+	//- successfully close the round through the submissions of the other nodes
+	updateAnswer(t, answerParams{fa: &fa, roundId: newRound,
+		answer: processedAnswer, from: fa.neil, isNewRound: false,
+		completesAnswer: true})
+
+	//- have the malicious node try to start another round repeatedly until the
+	//roundDelay is reached, making sure that it isn't successful
+	// Triggers a new round, since price deviation exceeds threshold
+	reportPrice = answer + 1
+	select {
```

Comment:
I think you might be able to do something like `require.Consistently(t, func() bool { return len(submissionReceived) == 0 })`.  You would need to make it a buffered channel of size 1, though.

Reply:
Do you mean something like 

```golang
	g := gomega.NewGomegaWithT(t)
	g.Eventually(submissionReceived).Should(gomega.Receive(),
		"chainlink failed to submit answer to FluxAggregator contract")
```
?

This is failing with assertion failures in `checkUpdateAnswer`:

```
    flux_monitor_simulated_blockchain_test.go:166: 
        	Error Trace:	flux_monitor_simulated_blockchain_test.go:166
        	            				flux_monitor_simulated_blockchain_test.go:317
        	Error:      	Should be true
        	Test:       	TestFluxMonitorAntiSpamLogic
        	Messages:   	SubmissionReceived log has wrong answer
    flux_monitor_simulated_blockchain_test.go:168: 
        	Error Trace:	flux_monitor_simulated_blockchain_test.go:168
        	            				flux_monitor_simulated_blockchain_test.go:317
        	Error:      	Not equal: 
        	            	expected: 0x2
        	            	actual  : 0x1
        	Test:       	TestFluxMonitorAntiSpamLogic
        	Messages:   	SubmissionReceived log has wrong round
    flux_monitor_simulated_blockchain_test.go:178: 
        	Error Trace:	flux_monitor_simulated_blockchain_test.go:178
        	            				flux_monitor_simulated_blockchain_test.go:317
        	Error:      	"[]" should have 1 item(s), but has 0
        	Test:       	TestFluxMonitorAntiSpamLogic
        	Messages:   	FluxAggregator did not emit correct NewRound log
```

I think the issue may be that [`Receive`](http://onsi.github.io/gomega/#receive) doesn't block.